src/components/ClickableStyle/ClickableStyle.tsx

```diff
@@ -0,0 +1,134 @@
+import clsx from 'clsx';
+import React, { MouseEventHandler, ReactNode } from 'react';
+import styles from './ClickableStyle.module.css';
+import Icon from '../Icon';
+
+export type ClickableStyleProps<IComponent extends React.ElementType> = {
+  /**
+   * Visually hidden clickable text (but text is still accessible to assistive technology).
+   * This overrides `children` for screen readers
+   */
+  'aria-label'?: string;
+  /**
+   * CSS class names that can be appended to the component
+   */
+  className?: string;
+  /**
+   * Disables the field and prevents editing the contents
+   */
+  disabled?: boolean;
+  /**
+   * ClickableStyle reference
+   */
+  forwardRef?: any;
+  /**
+   * Toggles clickable that fills the full width of its container
+   */
+  fullWidth?: boolean;
+  /**
+   * Loading state passed down from higher level used to trigger loader and text change
+   */
+  loading?: boolean;
+  /**
+   * On click handler for component
+   */
+  onClick?: MouseEventHandler;
+  /**
+   * Available size variations for the clickable
+   */
+  size?: 'sm' | 'md' | 'lg';
+  /**
+   * The visible clickable children
+   */
+  children: string | ReactNode;
+  /**
+   * Determines type of clickable
+   * - **button** The clickable is a clickable button.
+   * - **submit** The clickable is a submit clickable (submits form data).
+   * - **reset** The clickable is a reset clickable (resets the form-data to its initial values)
+   */
+  type?: 'button' | 'reset' | 'submit';
```

Comment:
The native HTML attributes are already included in the prop intersection `& React.ComponentProps<IComponent>`, so we don't _have_ to specify them here. It'd be more for documentation purposes then typing

Given that, wonder if some of these should only be declared in the `Button` component? specifically:
- disabled
- loading
- onClick
- type

don't seem relevant to `<a>` tags.

Reply:
I do wonder if that's making too many assumptions about what components might be passed into `ClickableStyle`, but I can't think of anything aside from `NavLink` and `Link` from react-router. I'll leave `onClick` though since there could be other logic in there for links (like analytics).